Source/WebCore/dom/Range.cpp

```diff
@@ -317,95 +319,111 @@ ExceptionOr<RefPtr<DocumentFragment>> Range::processContents(ActionType action)
+
+        RefPtr<Node> leftContents;
+        if (&originalStart.container() != commonRoot && commonRoot->contains(&originalStart.container())) {
+            auto firstResult = processContentsBetweenOffsets(action, nullptr, &originalStart.container(), originalStart.offset(), originalStart.container().length());
+            auto secondResult = processAncestorsAndTheirSiblings(action, &originalStart.container(), ProcessContentsForward, WTFMove(firstResult), commonRoot.get());
+            // FIXME: A bit peculiar that we silently ignore the exception here, but we do have at least some regression tests that rely on this behavior.
+            if (!secondResult.hasException())
+                leftContents = secondResult.releaseReturnValue();
+        }
 
-    RefPtr<Node> rightContents;
-    if (&endContainer() != commonRoot && commonRoot->contains(&originalEnd.container())) {
-        auto firstResult = processContentsBetweenOffsets(action, nullptr, &originalEnd.container(), 0, originalEnd.offset());
-        auto secondResult = processAncestorsAndTheirSiblings(action, &originalEnd.container(), ProcessContentsBackward, WTFMove(firstResult), commonRoot.get());
-        // FIXME: A bit peculiar that we silently ignore the exception here, but we do have at least some regression tests that rely on this behavior.
-        if (!secondResult.hasException())
-            rightContents = secondResult.releaseReturnValue();
-    }
+        RefPtr<Node> rightContents;
+        if (&endContainer() != commonRoot && commonRoot->contains(&originalEnd.container())) {
+            auto firstResult = processContentsBetweenOffsets(action, nullptr, &originalEnd.container(), 0, originalEnd.offset());
+            auto secondResult = processAncestorsAndTheirSiblings(action, &originalEnd.container(), ProcessContentsBackward, WTFMove(firstResult), commonRoot.get());
+            // FIXME: A bit peculiar that we silently ignore the exception here, but we do have at least some regression tests that rely on this behavior.
+            if (!secondResult.hasException())
+                rightContents = secondResult.releaseReturnValue();
+        }
+
+        // delete all children of commonRoot between the start and end container
+        RefPtr<Node> processStart = childOfCommonRootBeforeOffset(&originalStart.container(), originalStart.offset(), commonRoot.get());
+        if (processStart && &originalStart.container() != commonRoot) // processStart contains nodes before m_start.
+            processStart = processStart->nextSibling();
+        RefPtr<Node> processEnd = childOfCommonRootBeforeOffset(&originalEnd.container(), originalEnd.offset(), commonRoot.get());
+
+        // Collapse the range, making sure that the result is not within a node that was partially selected.
+        if (action == Extract || action == Delete) {
+            if (partialStart && commonRoot->contains(partialStart.get())) {
```

Comment:
`if (commonRoot->contains(partialStart.get())) {`
or
`if (partialStart && commonRoot->contains(*partialStart)) {`

Reply:
Opted for the latter.